service/controller/resource/tcnp/template/template_main_security_groups.go

```diff
@@ -13,6 +13,12 @@ const TemplateMainSecurityGroups = `
         FromPort: 22
         ToPort: 22
         CidrIp: {{ .SecurityGroups.ControlPlane.VPC.CIDR }}
+      -
+        Description: Allow traffic from control plane CIDR to 2049 for NFS access.
+        IpProtocol: tcp
+        FromPort: 2049
+        ToPort: 2049
+        CidrIp: {{ .SecurityGroups.ControlPlane.VPC.CIDR }}
```

Comment:
this does not look right

Why we would need to allow NFS port from the Management Cluster (COntrol plane cluster) ?

Reply:
I would expect we need to allow a range of the Worklaod cluster itself as it probably  goes via internal VPC routing

Reply:
You're right, I wanted to change that to pickup the VPC CIDR from the WC. Will change that

Reply:
Hmm it seems this is not trivial since we're only adding chunks of smaller ranges into it. Maybe I'm wrong but I don't know what would be the best way here.

Reply:
Just thinking it loud 🤔 

How do we ensure we have always up to date VPC CIDR's in our node pools? EFS plugin pulls the NFS mount once it can resolve filesystem fs-xxxxxxxx.efs.my-region.amazonaws.com. 

The thing is you create EFS and you can choose where you want to locate it and you get the IP. You could then potentially mount EFS in a different AZ, so the security group has to allow NFS inbound.

Reply:
My guess would be that we need to add the Tenant cluster  Guest cluster VPC main VPC range and that should be enough if the traffic is anyway going  VPC Endpoint -> TCNP node

I don't think we need to allow the traffic between tcnp nodes so adding additional VPC cidr that we add via VPCCidrAssociation is not necessary.

But that should be easy to test as well if you create more separate node pools